### PR TITLE
fix(dms/rabbitmq): fix issues with resizing RabbitMQ caused by API changes

### PR DESCRIPTION
**Pull request by chengxiangdong**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run TestAccDmsRabbitmqInstances'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run TestAccDmsRabbitmqInstances -timeout 360m -parallel 4
=== RUN   TestAccDmsRabbitmqInstances_basic
=== PAUSE TestAccDmsRabbitmqInstances_basic
=== RUN   TestAccDmsRabbitmqInstances_withEpsId
=== PAUSE TestAccDmsRabbitmqInstances_withEpsId
=== RUN   TestAccDmsRabbitmqInstances_compatible
=== PAUSE TestAccDmsRabbitmqInstances_compatible
=== RUN   TestAccDmsRabbitmqInstances_single
=== PAUSE TestAccDmsRabbitmqInstances_single
=== CONT  TestAccDmsRabbitmqInstances_basic
=== CONT  TestAccDmsRabbitmqInstances_compatible
=== CONT  TestAccDmsRabbitmqInstances_single
=== CONT  TestAccDmsRabbitmqInstances_withEpsId
--- PASS: TestAccDmsRabbitmqInstances_compatible (1239.81s)
--- PASS: TestAccDmsRabbitmqInstances_withEpsId (1781.24s)
--- PASS: TestAccDmsRabbitmqInstances_single (2273.27s)
--- PASS: TestAccDmsRabbitmqInstances_basic (2635.70s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       2635.785s
```

```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run TestAccKafkaInstance'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run TestAccKafkaInstance -timeout 360m -parallel 4
=== RUN   TestAccKafkaInstancesDataSource_basic
--- PASS: TestAccKafkaInstancesDataSource_basic (599.94s)
=== RUN   TestAccKafkaInstance_basic
=== PAUSE TestAccKafkaInstance_basic
=== RUN   TestAccKafkaInstance_withEpsId
=== PAUSE TestAccKafkaInstance_withEpsId
=== RUN   TestAccKafkaInstance_compatible
=== PAUSE TestAccKafkaInstance_compatible
=== RUN   TestAccKafkaInstance_newFormat
=== PAUSE TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_basic
=== CONT  TestAccKafkaInstance_compatible
=== CONT  TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_withEpsId
--- PASS: TestAccKafkaInstance_withEpsId (605.65s)
--- PASS: TestAccKafkaInstance_compatible (996.82s)
=== RUN   TestAccKafkaInstance_withEpsId
=== PAUSE TestAccKafkaInstance_withEpsId
=== RUN   TestAccKafkaInstance_compatible
=== PAUSE TestAccKafkaInstance_compatible
=== RUN   TestAccKafkaInstance_newFormat
=== PAUSE TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_basic
=== CONT  TestAccKafkaInstance_compatible
=== CONT  TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_withEpsId
--- PASS: TestAccKafkaInstance_withEpsId (605.65s)
--- PASS: TestAccKafkaInstance_compatible (996.82s)
--- PASS: TestAccKafkaInstance_newFormat (2269.91s)
--- PASS: TestAccKafkaInstance_basic (2693.77s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       3293.952s
```

### CODE REVIEW

File: huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go [hunk 1]
```diff
@@ -393,21 +399,21 @@ func resourceDmsRabbitmqInstanceRead(_ context.Context, d *schema.ResourceData,
...
-		logp.Printf("[WARN] error fetching tags of DMS rabbitmq instance (%s): %s", d.Id(), err)
-		e := fmtp.Errorf("error fetching tags of DMS rabbitmq instance (%s): %s", d.Id(), err)
+		log.Printf("[WARN] error fetching tags of DMS RabbitMQ instance (%s): %s", d.Id(), err)
+		e := fmt.Errorf("error fetching tags of DMS RabbitMQ instance (%s): %s", d.Id(), err)
 		mErr = multierror.Append(mErr, e)
 	}
 
 	if mErr.ErrorOrNil() != nil {
-		return fmtp.DiagErrorf("Error setting attributes for DMS rabbitmq instance: %s", mErr)
+		return diag.Errorf("failed to set attributes for DMS kafka instance: %s", mErr)
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go]: change  kafka to RabbitMQ

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 2]
```diff
@@ -872,37 +872,43 @@ func resourceDmsKafkaInstanceUpdate(ctx context.Context, d *schema.ResourceData,
...
-	opts := instances.ResizeInstanceOpts{}
+	resizeOpts := instances.ResizeInstanceOpts{}
 	if _, ok := d.GetOk("product_id"); ok {
-		if engineType == "kafka" {
-			product, err := getKafkaProductDetail(cfg, d)
+		if engineType == engineKafka {
+			product, err := getKafkaProductDetails(cfg, d)
 			if err != nil {
-				return fmt.Errorf("change storage_space failed, error querying product detail: %s", err)
+				return fmt.Errorf("failed to resize Kafka instance, query product details error: %s", err)
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go:885]: Failed to failed

File: huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go (hunk 1 above)
Comment (chengxiangdong) [reply huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go [hunk 3]
```diff
@@ -337,27 +341,28 @@ func resourceDmsRabbitmqInstanceCreate(ctx context.Context, d *schema.ResourceDa
...
-	availableZoneIDs := v.AvailableZones
-	availableZoneCodes, err := getAvailableZoneCodeByID(config, region, availableZoneIDs)
+	azIDs := v.AvailableZones
+	asCodes, err := getAvailableZoneCodeByID(cfg, region, azIDs)
 	mErr := multierror.Append(nil, err)
 
 	d.SetId(v.InstanceID)
 	mErr = multierror.Append(mErr,
-		d.Set("region", config.GetRegion(d)),
+		d.Set("region", cfg.GetRegion(d)),
```
Comment (hmbig2) [inline huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go]:
```suggestion
		d.Set("region", region),
```

Comment (chengxiangdong) [reply huaweicloud/services/dms/resource_huaweicloud_dms_rabbitmq_instance.go]: done

**Pull request merged**